phac-nml/irida-next
# Workflow Execution: Change state to enum

## What does this PR do and why?
This PR changes workflow execution state from type string to enum. It also sets the state to `running` when the returned state is `RUNNING` and removes the state `queued`.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
1. With workflow executions stored in your database, run `bin/rails db:migrate`. Ensure the migration is successful
2. Through the UI, check the workflow executions page displays the expected states per WE
3. Run a workflow execution containing a large number of samples. After sometime in the `submitted` state, ensure the WE state eventually changes to `running`. Afterwards, check that the WE completes.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
